app/controllers/users_controller.rb

```diff
@@ -1,4 +1,32 @@
 class UsersController < ApplicationController
   def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
+    if @user.nil?
+      render file: "public/404.html"
+    end
+
+  end
+
+  def create
+    @user = User.new user_params
+
+    if @user.save
+      flash[:success] = t "user_signup.success-signup"
+      redirect_to @user
+    else
+      flash[:warning] = t "user_signup.warning-signup"
```

Comment:
flash: tiếp tục xuất hiện tin nhắn khi thực hiện 1 action tiếp theo tại view mà mình mong muốn hiển thị. Vì thế nó thường được sử dụng khi muốn redirect đến 1 action khác thông qua hàm redirect_to
flash.now : tin nhắn hiển thị ở trang mà bạn đang thực hiện render

Khi em thử dùng thì khi mà sử dụng users/new để đăng ký lỗi 1 tai khoan thì sẽ được redirect đến users/ đồng thời xuất hiện thông báo lôi. tiếp tục nhấn reset trang /users/new thì vẫn báo lỗi đăng ký. Tuy nhiên flash.to thì không như vậy, k có tin nhắn báo lỗi nào dư thừa.